DirectEmployers/MyJobs
# [PD-3776] App for API key management.

## What's this PR do?
Introduces API key management.
Right now it runs as an app on secure, but it is designed to run as it's own app (hence the new database). 

This adds two types of keys for use with our APIs:
- Tokens. Right now we have no way of issuing these to users (this is temporarily a feature, not a bug). They belong to organizations, not specific users. 
- Temporary tokens. Good for 1 hour, eventually we will have permissions associated with these. This exists so organizations can hand out temporary, limited API access.


I didn't write endpoint documentation yet because I don't know where @darrint and @tejaswinigutti31 are going with it, and I think I'm better conforming to their standards rather than making my own.

## Where should the reviewer start?
The custom authentication backend for temporary keys is in `api_management/auth_backends.py`. Otherwise, most of the magic happens in `api_management/api/`.


## How should this be manually tested?
Check out the new endpoints for:
- Creating temporary API keys
- Listing existing, un-expired API keys
- Deleting API keys (even expired ones) 

You'll have to make your own endpoint for testing the TempTokenAuth, which is the whole point of creating/listing/destroying these keys.


## PR Checklist

### ~~Front-End~~
[removed by darrint]


### Separation of Concerns


### Tests


### Migrations
